source/module_base/math_sphbes.cpp

```diff
@@ -369,30 +369,31 @@ void Sphbes::Spherical_Bessel_Roots
     Sphbes::Spherical_Bessel(msh, r, 1, l, jl);
 
     int n=0;
-    for (int i=0; i<msh && n<num; i++)
+    for (int i=0; i<msh-1 && n<num; i++)
     {
         if (jl[i]*jl[i+1] < 0.0)
         {
             double y_1 = jl[i];
             double y_2 = jl[i+1];
             double x_1 = r[i];
             double x_2 = r[i+1];
-            double acc = std::fabs( y_2 - y_1 );
+            double acc = std::fabs(y_2 - y_1);
+
+            const int grid=100;
+            double *rad = new double[grid];
+            double *jl_new = new double[grid];
             while (acc > epsilon)
             {
-                double *rad = new double[100];
-                double *jl_new = new double[100];
-
                 // if not enough accurate, divide again.
-                const double delta2 = (x_2 - x_1)/99.0;
-                for (int j=0;j<100;j++)
+                const double delta2 = (x_2 - x_1)/(grid-1);
+                for (int j=0;j<grid;j++)
                 {
                     rad[j] = x_1 + j*delta2;
                 }
-                Sphbes::Spherical_Bessel(100,rad,1,l,jl_new);
+                Sphbes::Spherical_Bessel(grid,rad,1,l,jl_new);
 
                 int j=0;
-                for (;j<100;j++)
+                for (;j<grid-1;j++)
```

Comment:
I try to understand this PR that the out of bound only happened here?

Reply:
And also line 372